presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1188,8 +1193,25 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
         HiveTableHandle handle = checkType(tableHandle, HiveTableHandle.class, "tableHandle");
         HiveTableLayoutHandle layoutHandle = checkType(tableLayoutHandle, HiveTableLayoutHandle.class, "tableLayoutHandle");
 
-        for (HivePartition hivePartition : getOrComputePartitions(layoutHandle, session, tableHandle)) {
-            metastore.dropPartitionByName(handle.getSchemaName(), handle.getTableName(), hivePartition.getPartitionId());
+        Optional<Table> table = metastore.getTable(handle.getSchemaName(), handle.getTableName());
+        if (!table.isPresent()) {
+            throw new TableNotFoundException(handle.getSchemaTableName());
+        }
+
+        if (table.get().getPartitionKeysSize() == 0) {
+            // only delete data for managed tables
+            if (table.get().getTableType().equals(TableType.MANAGED_TABLE.toString())) {
```

Comment:
This should throw a `PrestoException(NOT_SUPPORTED, "Cannot delete from non-managed Hive table")`